### ISSUE TITLE
Overriding operationId programmatically in Azure Function

### ISSUE THREAD

**Issue by garoplin** [microsoft/ApplicationInsights-Java#1903]
**Is your feature request related to a problem? Please describe.**

We're trying to connect via distributed tracing (operationId) some azure services. They **don't** communicate over HTTP. The first one is sending message to Azure Storage Queue with attached operationId from the current context (
```ThreadContext.getRequestTelemetryContext().getHttpRequestTelemetry().getContext().getOperation().getId()``
). 
The second service is Azure Function that is triggered by this Storage Queue message. On the beginning of the processing, we're trying override operationId received in message body with the use of TelemetryClient or ThreadContext so it's gonna be populated to all further requests/traces:
1:
```
ThreadContext.getRequestTelemetryContext().getHttpRequestTelemetry().getContext().getOperation().setParentId(event.getData().getOperationId());
```

2:
```
telemetryClient.getContext().getOperation().setId(event.getData().getOperationId());
telemetryClient.flush();
```

However, none of this seems to be working. We're not able to override it, so the link between two services is broken and the dependencies are not connected in the distributed tracing tree.

Do you know how to make this work?

**Additional context**
The Azure Function is written in Java 11, and we're using application insights capabilities via Environment variables, as described in documentation:
1. ApplicationInsightsAgent_EXTENSION_VERSION=~2
2. XDT_MicrosoftApplicationInsights_Java=1

In the code we're using ApplicationInsights Core (https://mvnrepository.com/artifact/com.microsoft.azure/applicationinsights-core/2.6.3) and ApplicationInsights Web (https://mvnrepository.com/artifact/com.microsoft.azure/applicationinsights-web/2.6.3)

**Comment by trask** [comment]
Hi @garoplin

> The first one is sending message to Azure Storage Queue

Can you try running Application Insights Java 3.2.0 on this service?

The reason it might help is that 3.2.0 includes auto-instrumentation of Azure Storage Queues SDK (version 12.9.0+), and so it should be setting the correct distributed tracing header, which then Azure Functions picks up, and since you're using the Azure Functions Application Insights Java integration, that distributed trace should be automatically set as the parent for your Azure Functions call.

Let us know if this works or not, and we can look into it further.

**Comment by rangarnow** [comment]
Hi @trask, thanks for your response, it was super helpful!

We found that now App service and Azure function link is presented in Application Insights and the two traces looks the same and share operation id, so it's great.

I saw however that before we had the name of the azure function visible in Application Insights tracking and every invocation was visible under that function, so relation parent -> child were kept.

Currently we see that Azure function calls are presented on the same level as App Service calls and it's hard to distinguish when Azure function begins. Also the name of Azure function is not anymore presented (and perhaps other events happenning before `context.makeCurrent()`). It would be perfect to see Azure function as a part of trace but as a separate invocation (not next to App service calls) and with it's own name on the top. Something like this:

![image](https://user-images.githubusercontent.com/92714803/137742757-cebb866e-7d3b-4c4e-8a54-af15bfdebedd.png)
